docs/java/activities.md

```diff
@@ -4,24 +4,65 @@ title: Activities in Java
 sidebar_label: Activities
 ---
 
-Activities can be invoked during Workflow execution. Similar to Workflows, Activities in Temporal
-Java SDK programming model are classes which implement an Activity Interface:
+## What is an Activity?
+
+Having fault tolerant code that maintains state is useful.
+But in most practical applications, a Workflow will be expected to act upon the external world.
+Temporal supports such externally-facing code in the form of Activities.
+
+An Activity is essentially a function that can execute any code like DB updates or service calls.
+The Workflow is not allowed to directly call any external APIs; it can do this only through Activities.
+The Workflow is essentially an orchestrator of Activities.
+Activities that fail or timeout will be retried; therefore they should be written with care to be idempotent.
+### Activity interface
+
+Similar to Workflows, Activities in Temporal Java SDK programming model are classes which implement an Activity Interface:
+
+```java
+@ActivityInterface
+public interface GreetingActivities {
+    String composeGreeting(String greeting, String name);
+}
+```
+
+An Activity interface must be annotated with the `@ActivityInterface`.
+Each method that belongs to an `@ActivityInterface` defines a separate Activity type.
+You can annotate each method in the Activity interface with the `@ActivityMethod` annotation, but this is completely optional.
 
 ```java
 @ActivityInterface
 public interface GreetingActivities {
+    @ActivityMethod(name=composeHello)
     String composeGreeting(String greeting, String name);
 }
 ```
 
-Activity interface must be annotated with the `@ActivityInterface`. You can annotate each method in the
-Activity interface with the `@ActivityMethod` annotation, but this is completely optional.
-Note that the `@ActivityMethod` annotation has a `name` parameter which can be used to define
-the Activity type.
-If not specified, the method name (with the first letter capitalized) is used by default, so in 
-the above example it would be "ComposeGreeting".
+Note that the `@ActivityMethod` annotation has a `name` parameter which can be used to define the Activity type.
+If not specified, the method name (with the first letter capitalized) is used by default, so in the above example it would be `ComposeHello`. If the annotation did not specify the name then the Activity Type would be `ComposeGreeting`.
+
+### Activity implementation
 
-### Calling Activities inside Workflows
+An Activity implementation is just a normal [POJO](https://en.wikipedia.org/wiki/Plain_old_Java_object).
+The `out` stream is passed as a parameter to the constructor to demonstrate that the Activity object can have any dependency.
```

Comment:
This is because when registering an activity with worker you use the activity instance, and not the type like with Workflows. maybe we can add

Reply:
added